src/core/metrics/exporter/retry_socket.rs

```diff
@@ -0,0 +1,103 @@
+//! A TCP Socket wrapper that reconnects automatically.
+
+use log::{info, warn};
+use std::fmt;
+use std::io;
+use std::io::Write;
+use std::net::TcpStream;
```

Comment:
It is better to use tokio async TcpStream because bob is built as an async app

Reply:
You can skip part with implementation of AsyncWrite/AsyncRead for RetrySocket, if it seems too complicated

Reply:
I am not sure that we do need this now because the code of this `RetrySocket` I found in dipstick - crate, which we used for metrics before. Maybe we should leave this task in backlog for now?